lib/hotel.rb

```diff
@@ -0,0 +1,92 @@
+require_relative 'room'
+require_relative 'reservation'
+
+module HotelSystem
+  class Hotel
+    attr_reader :rooms, :reservations
+    attr_accessor :amount_of_rooms
+    
+    def initialize(amount_of_rooms)
+      @amount_of_rooms = amount_of_rooms
+      @rooms = []
+      @reservations = []
+    end
+    
+    def create_rooms
+      cost = 200
+      room_number = 0
+      @amount_of_rooms.times do
+        room_number += 1 
+        room = HotelSystem::Room.new(room_number: room_number, cost: cost)
+        rooms.push(room)
+      end
+      
+    end
+    
+    def list_rooms
+      create_rooms
+      rooms
+    end
+    
+    def make_reservation(start_date, end_date)
+      if end_date < start_date
+        raise ArgumentError.new("Error, the end date you entered #{end_date}, occurs before reservation start date, #{start_date}")
+      elsif end_date == start_date
+        raise ArgumentError.new("Error, reservation has a one night stay minimum.")
+      end
+      
+      reservation_id = rand(100..500)
+      
+      available_rooms = find_available_rooms(start_date, end_date)
+      
+      if available_rooms.empty?
+        raise StandardError.new("No available rooms.")
+      end
+      
+      room = available_rooms.first
+      reservation = HotelSystem::Reservation.new(reservation_id: reservation_id, room: room, start_date: start_date, end_date: end_date)
```

Comment:
Ruby will be looking for names locally first, prioritizing local names, since they're all in the same module, there's no need to namespace `HotelSystem::Reservation`. `Reservation` should just work! Same with the rest of the namespacing in the code.